plugin/grails-app/controllers/grails/plugin/springsecurity/ui/RegisterController.groovy

```diff
@@ -118,79 +136,116 @@ class RegisterController extends AbstractS2UiController implements GrailsConfigu
+			if (forgotPasswordExtraValidation && forgotPasswordExtraValidation.size() > 0) {
+				redirect uri: generateLink('securityQuestions', [username: forgotPasswordCommand.username])
+			} else {
+				if (requireForgotPassEmailValidation) {
+					processForgotPasswordEmail(forgotPasswordCommand, user)
+				} else {
+					redirect uri: processForgotPasswordEmail(forgotPasswordCommand, user)
+				}
+			}
+		}.invalidToken {
+			flash.message = "Invalid Form Submission"
+			redirect(controller: "login", action: "auth")
+		}
 	}
 
-	def forgotPassword(ForgotPasswordCommand forgotPasswordCommand) {
 
-		if (!request.post) {
-			return [forgotPasswordCommand: new ForgotPasswordCommand()]
-		}
-
-		if (forgotPasswordCommand.hasErrors()) {
-			return [forgotPasswordCommand: forgotPasswordCommand]
+	def securityQuestions(){
+		if (forgotPasswordExtraValidation && forgotPasswordExtraValidation.size() > 0) {
+				def user = findUserByUsername(params.username)
+				SecurityQuestionsCommand sc = new SecurityQuestionsCommand()
+				try {
+					sc.username = params.username
+					if (!request.post) {
+						render(view:"securityQuestions", model: [securityQuestionsCommand:sc,user:user,forgotPasswordExtraValidation:forgotPasswordExtraValidation,validationUserLookUpProperty:validationUserLookUpProperty])
+					} else {
+						withForm {
+							def rtn = uiRegistrationCodeStrategy.validateForgotPasswordExtraSecurity(params, user, forgotPasswordExtraValidation, validationUserLookUpProperty)
+							if (!rtn[0]) {
+								sc.validations = rtn[1]
+								render(view: "securityQuestions", model: [securityQuestionsCommand: sc, user: user, forgotPasswordExtraValidation: forgotPasswordExtraValidation, validationUserLookUpProperty: validationUserLookUpProperty])
+							} else {
+								ForgotPasswordCommand fc = new ForgotPasswordCommand()
+								fc.username = sc.username
+								if (requireForgotPassEmailValidation) {
+									render(view: "forgotPassword", processForgotPasswordEmail(fc, user))
+								} else {
+									redirect uri: processForgotPasswordEmail(fc, user)
+								}
+							}
+						}.invalidToken {
+							flash.message = "Invalid Form Submission"
+							redirect(controller: "login", action: "auth")
+						}
+					}
+				} catch (Exception e) {
+					log.error e
+					render(view: "securityQuestions", model: [securityQuestionsCommand: sc,user: user, forgotPasswordExtraValidation: forgotPasswordExtraValidation, validationUserLookUpProperty: validationUserLookUpProperty])
+				}
+			} else {
+				redirect uri: generateLink('register')
+			}
 		}
 
-		def user = findUserByUsername(forgotPasswordCommand.username)
-		if (!user) {
-			forgotPasswordCommand.errors.rejectValue 'username', 'spring.security.ui.forgotPassword.user.notFound'
-			return [forgotPasswordCommand: forgotPasswordCommand]
-		}
 
+	protected def processForgotPasswordEmail(forgotPasswordCommand,user){
 		String email = uiPropertiesStrategy.getProperty(user, 'email')
```

Comment:
Line #216: `String email = uiPropertiesStrategy.getProperty(user, 'email')` should be moved inside of the `if` statement at line 217, `if(requireForgotPassEmailValidation) {` to avoid the error `ERROR --- [io-8080-exec-10] g.p.s.ui.SpringSecurityUiService         : Attempted to read non-existent property email in User(username:me)` if email is not being used.

It should look something like:

```
	protected def processForgotPasswordEmail(forgotPasswordCommand,user){
		if(requireForgotPassEmailValidation) {
		        String email = uiPropertiesStrategy.getProperty(user, 'email')
			if (!email) {
.
.
.

```

Reply:
This will be fixed in my next pull request